### PR TITLE
ShoppingCart impl,CartItem impl + refactor dto to record instead of c…

### PR DESCRIPTION
**Pull request by artemsa223**
…lass

### CODE REVIEW

File: src/main/java/com/example/demo/model/CartItem.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
```
Comment (koval-kn) [inline src/main/java/com/example/demo/model/CartItem.java]: dont use data on entities, as it causes bugs. use getter setter annotations instead

File: src/main/java/com/example/demo/model/ShoppingCart.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import jakarta.persistence.Table;
+import java.util.Set;
+import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
+
+@Data
```
Comment (koval-kn) [inline src/main/java/com/example/demo/model/ShoppingCart.java]: same

File: src/main/java/com/example/demo/model/CartItem.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    @JoinColumn(name = "shopping_cart_id", nullable = false)
+    private ShoppingCart shoppingCart;
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "book_id", nullable = false)
+    private Book book;
+    @NotNull
+    private int quantity;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
```
Comment (koval-kn) [inline src/main/java/com/example/demo/model/CartItem.java]:
```suggestion
```

File: src/main/java/com/example/demo/model/ShoppingCart.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+
+    @OneToMany(mappedBy = "shoppingCart",fetch = FetchType.LAZY)
+    private Set<CartItem> cartItems;
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+
+
```
Comment (koval-kn) [inline src/main/java/com/example/demo/model/ShoppingCart.java]:
```suggestion
```

File: src/main/java/com/example/demo/model/ShoppingCart.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+@Table(name = "shopping_carts")
+public class ShoppingCart {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @OneToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+
+    @OneToMany(mappedBy = "shoppingCart",fetch = FetchType.LAZY)
```
Comment (koval-kn) [inline src/main/java/com/example/demo/model/ShoppingCart.java]:
```suggestion
    @OneToMany(mappedBy = "shoppingCart", fetch = FetchType.LAZY)
```

**Review by koval-kn — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**